Library/Homebrew/os/linux/elf.rb

```diff
@@ -68,28 +68,30 @@ def binary_executable?
     elf_type == :executable
   end
 
-  def with_interpreter?
-    return @with_interpreter if defined? @with_interpreter
+  def interpreter
+    return @interpreter if defined? @interpreter
 
-    @with_interpreter = if binary_executable?
-      true
-    elsif dylib?
-      if HOMEBREW_PATCHELF_RB
-        begin
-          patchelf_patcher.interpreter.present?
-        rescue PatchELF::PatchError => e
-          opoo e unless e.to_s.start_with? "No interpreter found"
-          false
-        end
-      elsif which "readelf"
-        Utils.popen_read("readelf", "-l", to_path).include?(" INTERP ")
-      elsif which "file"
-        Utils.popen_read("file", "-L", "-b", to_path).include?(" interpreter ")
-      else
-        raise "Please install either readelf (from binutils) or file."
+    if HOMEBREW_PATCHELF_RB
+      begin
+        patchelf_patcher.interpreter
+      rescue PatchELF::PatchError => e
+        opoo e unless e.to_s.start_with? "No interpreter found"
+        nil
       end
+    elsif (patchelf = DevelopmentTools.locate "patchelf")
+      # the check is probably redundant, it is already done by #keg_relocate
+      # adding it as a safety precaution. And maybe for something else to use
+      # in future.
```

Comment:
You can remove this comment.

Reply:
oops, that was meant to be discussed here.  sure i'll remove the comment, but should we do that check ?

Reply:
Yeah, this `interpreter` could potentially be called from a different location other than `keg_relocate` that doesn't guarantee that `patchelf` is installed.